### PR TITLE
chore: update deps

### PR DESCRIPTION
**Pull request by alanpoulain**

### CODE REVIEW

File: src/BookStore/Infrastructure/ApiPlatform/State/Provider/BookItemProvider.php [hunk 1]
```diff
@@ -13,6 +13,9 @@
 use App\Shared\Application\Query\QueryBusInterface;
 use Symfony\Component\Uid\Uuid;
 
+/**
+ * @template-implements ProviderInterface<BookResource>
```
Comment (mtarld) [inline src/BookStore/Infrastructure/ApiPlatform/State/Provider/BookItemProvider.php:17]: Isn't `@implements` enough?

File: src/BookStore/Infrastructure/ApiPlatform/State/Processor/UpdateBookProcessor.php [hunk 2]
```diff
@@ -33,9 +33,10 @@ public function __construct(
     public function process($data, Operation $operation, array $uriVariables = [], array $context = []): mixed
     {
         Assert::isInstanceOf($data, BookResource::class);
+        Assert::isInstanceOf($context['previous_data'], BookResource::class);
```
Comment (mtarld) [inline src/BookStore/Infrastructure/ApiPlatform/State/Processor/UpdateBookProcessor.php:36]:
```suggestion
        Assert::isInstanceOf($context['previous_data'] ?? null, BookResource::class);
```

Comment (alanpoulain) [reply src/BookStore/Infrastructure/ApiPlatform/State/Processor/UpdateBookProcessor.php:36]: Previous data should always be here, is it necessary?

File: src/BookStore/Infrastructure/ApiPlatform/State/Provider/BookItemProvider.php (hunk 1 above)
Comment (alanpoulain) [reply src/BookStore/Infrastructure/ApiPlatform/State/Provider/BookItemProvider.php]: Sure is!

File: src/BookStore/Infrastructure/ApiPlatform/State/Processor/UpdateBookProcessor.php (hunk 2 above)
Comment (mtarld) [reply src/BookStore/Infrastructure/ApiPlatform/State/Processor/UpdateBookProcessor.php:36]: Indeed, I think we can trust API Platform to not introduce BC breaks :wink:

File: src/BookStore/Infrastructure/ApiPlatform/State/Provider/BookCollectionProvider.php [hunk 3]
```diff
@@ -14,6 +14,9 @@
 use App\Shared\Application\Query\QueryBusInterface;
 use App\Shared\Infrastructure\ApiPlatform\State\Paginator;
 
+/**
+ * @template-implements ProviderInterface<BookResource>
```
[the remark this answers is not in the document]
Comment (alanpoulain) [reply src/BookStore/Infrastructure/ApiPlatform/State/Provider/BookCollectionProvider.php]: Sorry!

File: src/BookStore/Infrastructure/ApiPlatform/State/Provider/CheapestBooksProvider.php [hunk 4]
```diff
@@ -11,6 +11,9 @@
 use App\BookStore\Infrastructure\ApiPlatform\Resource\BookResource;
 use App\Shared\Application\Query\QueryBusInterface;
 
+/**
+ * @template-implements ProviderInterface<BookResource>
```
[the remark this answers is not in the document]
Comment (alanpoulain) [reply src/BookStore/Infrastructure/ApiPlatform/State/Provider/CheapestBooksProvider.php]: Done :slightly_smiling_face:

**Review by mtarld — APPROVED** [review]

**Pull request merged**